### PR TITLE
NETOBSERV-63 HPA

### PR DESCRIPTION
**Pull request by jpinsonneau**
add hpa spec for flow collector with MinReplicas, MaxReplicas and TargetCPUUtilizationPercentage
target goflow-kube Deployment

### CODE REVIEW

File: controllers/goflowkube/goflowkube_reconciler.go [hunk 1]
```diff
@@ -63,6 +69,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, desired *flowsv1alpha1.FlowC
 			newSVC := buildService(desired, r.OperatorNamespace)
 			r.createOrUpdate(ctx, oldSVC, newSVC, serviceKind)
 		}
+
+		//Delete or Create / Update Autoscaler according to HPA option
+		if desired.HPA == nil {
+			r.delete(ctx, oldASC, autoscalerKind)
```
Comment (jotak) [inline controllers/goflowkube/goflowkube_reconciler.go:75]: what if `oldASC` is nil? We probably shouldn't try to delete it then

File: controllers/goflowkube/goflowkube_reconciler.go [hunk 2]
```diff
@@ -63,6 +69,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, desired *flowsv1alpha1.FlowC
...
+		} else if oldASC == nil || autoScalerNeedsUpdate(oldASC.(*ascv1.HorizontalPodAutoscaler), desired) {
+			newASC := buildAutoScaler(desired, r.OperatorNamespace)
+			r.createOrUpdate(ctx, oldASC, newASC, autoscalerKind)
+		}
 	case daemonSetKind:
```
Comment (jotak) [inline controllers/goflowkube/goflowkube_reconciler.go:80]: in the daemonset case we should certainly delete the HPA to not leave any unused object out there

File: controllers/goflowkube/goflowkube_reconciler.go (hunk 1 above)
Comment (jpinsonneau) [reply controllers/goflowkube/goflowkube_reconciler.go:75]: also changed in daemonSetKind case for oldSVC

**Comment by jpinsonneau** [comment]
lgtm

**Pull request merged**